docs/specs/code/TRANSACTION_MEMO.md

```diff
@@ -0,0 +1,156 @@
+* [Download single file](https://dropbox.github.io/dropbox-sdk-js/Dropbox.html#filesDownload__anchor): Download a file from a user's Dropbox.
+* [Search individual file](https://dropbox.github.io/dropbox-sdk-js/Dropbox.html#filesSearch__anchor): Searches for files and folders. Note: Recent changes may not immediately be reflected in search results due to a short delay in indexing.
+* [Download folder as a ZIP file](https://dropbox.github.io/dropbox-sdk-js/Dropbox.html#filesDownloadZip__anchor): Download a folder from the user's Dropbox, as a zip file. The folder must be less than 20 GB in size and have fewer than 10,000 total files. The input cannot be a single file. Any single file must be less than 4GB in size.
+
+Using Dropbox will require to [create an app](https://www.dropbox.com/developers/reference/developer-guide). The user will have to connect his/her account and grant permissions, after which an [access token](https://www.dropbox.com/developers/reference/oauth-guide) will be generated. This access token must be safely stored locally in order to make authorized requests on behalf of the user (refer to the [encryption](https://github.com/Emurgo/yoroi-frontend/blob/737595fec5a89409aacef827d356c9a1605515c0/docs/specs/code/ENCRYPT.md) spec). 
+
+Note: API rate limits will apply in a [per user/app basis](https://www.dropbox.com/developers/reference/data-ingress-guide), so it shouldn't be a concern for Yoroi itself.
+
+## User flow
+
+The proposed user flow is the following:
+
+1. The user opens the send transaction dialog.
+2. The memo textarea is closed by default. The user can click *"Add memo"* label and one of the following scenarios occurs:
+   - If the user hasn't synced his external storage account yet, he/she will see the text "In order to add private memos to   transactions you need to link Yoroi to a Dropbox account." and a link to the settings page.
+   		- In the settings page, the user can click *"Connect Yoroi to Dropbox"*.
+   - If the user has already synced his external storage account, the memo textarea will be shown.
+3. If the user clicks *"Add memo"* again:
+   - If the textarea is empty, it will be closed.
+   - If it isn't empty, the user will see a dialog window with the text *"Do you want to discard this memo?"* and two buttons showing "Yes/No".
+     - If the user clicks *"No"*, the dialog closes and the memo textarea remains opened with the same content.
+     - If the user clicks *"Yes"*, the dialog window closes and the textarea become cleared.
+4. If user has entered the memo and makes a transaction, the memo will be saved to local storage and then to external storage. If there is a problem saving to external storage, then the user will see an error message: *"An error occurred while syncing with your Dropbox account. You can try to sync again or discard the memo"*. In that case, one of the following scenarios occurs:
```

Comment:
what is the format for localstorage? It isn't described anywhere. Do we save an indexdb table? As an object in `storage.local`?
Note: if we save in `storage.sync` then it will automatically sync with their browser storage  (but it sounds like it's not well supported on Brave and other Chrome forks)